File: 프로그래머스/lv3/43162. 네트워크/네트워크.py
def solution(n, computers):
    visited = [False for _ in range(n)]
    answer = 0
    for i in range(n):
        if not visited[i]:
            dfs(n,computers,i,visited)
            answer+=1
    return answer
def dfs(n,computers,start, visited):
    visited[start] = True
    for i in range(n):
        if visited[i] == False and computers[start][i] == 1:
            dfs(n,computers,i,visited)
            
    

File: 프로그래머스/3/43162. 네트워크/네트워크.java
import java.util.*;
class Solution {
    boolean[] visited;
    
    public int solution(int n, int[][] computers) {
        int answer = 0;
        visited = new boolean[computers.length];
        
        for(int i=0; i<n; i++){
            if(!visited[i]){
                DFS(computers,i);
                answer++;
            }
        }
        return answer;
    }
    void DFS(int[][] computers,int node){
        visited[node] = true;
        
        for(int j=0; j<computers.length; j++){
            if(node != j && computers[node][j] == 1 && visited[j] == false){
                DFS(computers,j);
            }
        }
    }
}